### DISCUSSION TITLE
Improve persistent marker api

### DISCUSSION THREAD

**Discussion by Inky-developer** [patrykandpatrick/vico#771] [Ideas]
Hi, 
First of all thanks for this very nice library!

I am trying to create a stacked bar graph which displays the type of each stacked bar segment within the graph to make it easier to interpret the data. I have managed to get quite close to the wanted result by subclassing `CartesianMarker`:
![image](https://github.com/patrykandpatrick/vico/assets/47245667/9a3ccc20-4ed7-4025-bf96-056f886964a5)

For reference, here is the code I am using for the markers:
<details>

```kotlin
class IconMarker(private val indicator: TextComponent, private val emojis: List<String>) : CartesianMarker {
    override fun draw(context: CartesianDrawContext, targets: List<CartesianMarker.Target>) {
        with(context) {
            targets.forEach { target ->
                when (target) {
                    is ColumnCartesianLayerMarkerTarget -> {
                        target.columns.zip(emojis).forEach { (column, emoji) ->
                            if (column.entry.y != 0f) {
                                drawIndicator(target.canvasX, column.canvasY, emoji)
                            }
                        }
                    }

                    else -> throw NotImplementedError("IconMarker only supports columns")
                }
            }
        }
    }

    private fun CartesianDrawContext.drawIndicator(x: Float, y: Float, emoji: String) {
        indicator.drawText(
            this,
            emoji,
            x,
            y
        )
    }
}
```

</details>

However, I want to display the markers in the middle of each segment instead of at the top. As far as I could figure out, that is not possible with the current api. 
A possible improvement would be to give information about the bounds of each bar segment.

Additionally, I think it is not quite ideal that we need to provide a map containing a persistent marker for every single x value. For my usecase I would prefer to set just on persistent marker that applies to every column.

**Comment by Gowsky** [comment]
Hello @Inky-developer,
>However, I want to display the markers in the middle of each segment instead of at the top. As far as I could figure out, that is not possible with the current api.
A possible improvement would be to give information about the bounds of each bar segment.

All the data needed to implement this is already there. You can check the adjacent y values, and the y range and bounds are included in `CartesianDrawContext`. However, we'll look into providing more information in the targets for simplicity.

>Additionally, I think it is not quite ideal that we need to provide a map containing a persistent marker for every single x value. For my usecase I would prefer to set just on persistent marker that applies to every column.

The API needs to be flexible and handle various use cases, and this one isn't so common. However, we'll look into adding an API that makes it easier to add a marker to every entry.

**Comment by Inky-developer** [comment]
Thank you for the quick response!
